lib/mini_racer/shared.rb

```diff
@@ -0,0 +1,380 @@
+# This code used to be shared in lib/mini_racer.rb
+# but was moved to the extension with https://github.com/rubyjs/mini_racer/pull/325.
+# So now this is effectively duplicate logic with C/C++ code.
+# Maybe one day it can be actually shared again between both backends.
+
+module MiniRacer
+
+  MARSHAL_STACKDEPTH_DEFAULT = 2**9-2
+  MARSHAL_STACKDEPTH_MAX_VALUE = 2**10-2
```

Comment:
FWIW, these are gone; the C code accepts `marshal_stack_depth` but otherwise ignores it.